efel/cppcore/cppcore.cpp

```diff
@@ -108,36 +108,48 @@ static void PyList_from_vectorstring(vector<string> input, PyObject* output) {
 }
 
 static PyObject*
-_getfeature(PyObject* self, PyObject* args, const string &type) {
+_getfeature(PyObject* self, PyObject* args, const string &input_type) {
   char* feature_name;
   PyObject* py_values;
 
   int return_value;
   if (!PyArg_ParseTuple(args, "sO!", &feature_name, &PyList_Type, &py_values)) {
+    PyErr_SetString(PyExc_TypeError, "Unexpected argument type provided.");
     return NULL;
   }
 
-  string feature_type = pFeature->featuretype(string(feature_name));
-
-  if (!type.empty() && feature_type != type){
-    PyErr_SetString(PyExc_TypeError, "Feature type does not match");
+  try {
+    string feature_type = pFeature->featuretype(string(feature_name));
+
+    if (!input_type.empty() && feature_type != input_type){  // when types do not match
+      PyErr_SetString(PyExc_TypeError, "Feature type does not match");
+      return NULL;
+    }
+
+    if (feature_type == "int") {
+      vector<int> values;
+      return_value = pFeature->getFeature<int>(string(feature_name), values);
+      PyList_from_vectorint(values, py_values);
+      return Py_BuildValue("i", return_value);
+    } else { // double
+      vector<double> values;
+      return_value = pFeature->getFeature<double>(string(feature_name), values);
+      PyList_from_vectordouble(values, py_values);
+      return Py_BuildValue("i", return_value);
```

Comment:
I think it should be "d" here

Reply:
That is actually the integer return code that has the following meaning at the moment
"if < 0 then failure, otherwise success."
The features are passed to python in the previous line

```python 
PyList_from_vectordouble(values, py_values);
```



It's a bit confusing.

Reply:
ha sorry 👍